python_on_whales/components/image/cli_wrapper.py

```diff
@@ -503,49 +506,133 @@ def pull(
         self,
         x: Union[str, Iterable[str]],
         quiet: bool = False,
+        stream_logs: bool = False,
         platform: Optional[str] = None,
-    ) -> Union[Image, List[Image]]:
+    ) -> Union[Image, List[Image], Iterable[Tuple[str, bytes]]]:
         """Pull one or more docker image(s)
 
         Alias: `docker.pull(...)`
 
         Parameters:
-            x: The image name(s) . Can be a string or a list of strings. In case of
-                list, multithreading is used to pull the images.
-                The progress bars might look strange as multiple
-                processes are drawing on the terminal at the same time.
+            x: Image name(s), can be a string or a list of strings.
+                In case a list is passed, multithreading is used to pull
+                the images.
             quiet: If you don't want to see the progress bars.
+            stream_logs: If `False` this function returns the pulled image(s).
+                If `True`, this function returns an `Iterable` of `Tuple[str, bytes]`
+                where the first element corresponds to the image name that a pull is being done for.
+                The second element is the log statement related to pull progress, as `bytes`.
+                You'll need to call `.decode()` if you want the logs as `str`.
+                See [the streaming guide](https://gabrieldemarmiesse.github.io/python-on-whales/user_guide/docker_run/#stream-the-output)
```

Comment:
The streaming guide is not very helpful (and can be even confusing here as it's not the same format). You can just remove this line. Same for push.